config.go

```diff
@@ -394,6 +416,38 @@ func (cfg *Config) unshadow() {
 	cfg.shadow = nil
 }
 
+func generateClusterSecret() ([]byte, error) {
+	encodedSecretLength := 64
+	secret := make([]byte, base64.StdEncoding.DecodedLen(encodedSecretLength))
+	_, err := crand.Read(secret)
+	if err != nil {
+		return nil, fmt.Errorf("Error reading from rand: %v", err)
+	}
+	encodedSecret := make([]byte, encodedSecretLength)
+	base64.StdEncoding.Encode(encodedSecret, secret)
+	return encodedSecret, nil
+}
+
+func clusterSecretToKey(secret []byte) (string, error) {
+	var key bytes.Buffer
+	key.WriteString("/key/swarm/psk/1.0.0/\n")
+	key.WriteString("/base64/\n")
+	key.Write(secret)
+
+	return key.String(), nil
+}
+
+func IsValidSecret(secret []byte) (bool, error) {
+	var isBase64 = regexp.MustCompile(`^[0-9A-Za-z+/]+$`).Match
```

Comment:
Base64 encoding has `=` sign for padding

Just do a `parseSecret` function that does https://golang.org/pkg/encoding/base64/#Encoding.DecodeString (and returns an error if it doesn't work, and the checks that the length is 32 before returning the byte slice. No need to regexp stuff ourselves.